pkg/k8sclient/k8sclient.go

```diff
@@ -268,3 +276,123 @@ func GetPodLogs(ctx context.Context, k8sClient kubernetes.Interface, namespace s
 	}
 	return logs, nil
 }
+
+func PortForward(ctx context.Context, namespace, serviceName string, servicePort, localhostPort int) error {
+
+	clientSet, err := ConnectToK8s()
+	if err != nil {
+		return err
+	}
+	svc, err := clientSet.CoreV1().Services(namespace).Get(ctx, serviceName, v1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	var podPort intstr.IntOrString
+	for _, port := range svc.Spec.Ports {
+		if port.Port == int32(servicePort) {
+			podPort = port.TargetPort
+		}
+	}
+
+	pods, err := clientSet.
+		CoreV1().
+		Pods(namespace).
+		List(ctx, v1.ListOptions{
+			LabelSelector: labels.SelectorFromSet(labels.Set(svc.Spec.Selector)).String(),
+		})
+	if err != nil {
+		return err
+	}
+
+	var servicePod *corev1.Pod
+	for _, pod := range pods.Items {
+		if pod.Status.Phase != corev1.PodRunning {
+			continue
+		}
+		servicePod = &pod
+		break
+	}
+
+	if servicePod == nil {
+		return fmt.Errorf("no running pods found for service %s/%s", namespace, serviceName)
+	}
+
+	var podPortNumber int32
+	for _, c := range servicePod.Spec.Containers {
+		for _, p := range c.Ports {
+			if p.ContainerPort == podPort.IntVal || p.Name == podPort.StrVal {
+				podPortNumber = p.ContainerPort
+			}
+		}
+	}
+
+	restConfig, err := GetK8sClientConfig()
+	if err != nil {
+		return err
+	}
+
+	transport, upgrader, err := spdy.RoundTripperFor(restConfig)
+	if err != nil {
+		return errors.Wrap(err, "create round tripper")
+	}
+
+	readyChan := make(chan struct{})
+
+	url := clientSet.
+		CoreV1().
+		RESTClient().
+		Post().
+		Resource("pods").
+		Namespace(namespace).
+		Name(servicePod.Name).
+		SubResource("portforward").
+		URL()
+
+	dialer := spdy.NewDialer(upgrader, &http.Client{Transport: transport}, "POST", url)
+	forwarder, err := portforward.New(dialer, []string{fmt.Sprintf("%d:%d", localhostPort, podPortNumber)}, ctx.Done(), readyChan, os.Stdout, os.Stderr)
```

Comment:
why do we port forward for a particular pod, not a service?

Reply:
should be localhostPort fixed, or we can dynamically check available one?

Reply:
I was not able to make it work with the service

Reply:
we can go with it, but pod is not something relaible, can be esily killed and restrarted. service is a logical unit